src/apps/service/apps/de/job_view.clj

```diff
@@ -142,3 +143,12 @@
     (verify-app-permission user app "read")
     (validate-hidden-inputs user (:version_id app))
     (format-app user app include-hidden-params?)))
+
+(defn get-app-version
+  "This service obtains an app description in a format that is suitable for building the job
+  submission UI."
+  [user app-id version-id]
+  (let [app (amp/get-app-version app-id version-id)]
+    (verify-app-permission user app "read")
+    (validate-hidden-inputs user (:version_id app))
+    (format-app user app false)))
```

Comment:
This `get-app-version` is not for listing an app, but for getting the "app description", which is what the UI uses to build the app launch form (it formats the param IDs required in job submissions, for example).

This function does the same thing as `get-app`, but `get-app` will only get the app launch "description" for the app's latest version, and `get-app-version` can get the description for any version.

The call to `amp/list-app-versions` is so that the `versions: [{version, version_id}, ...]` list is included in the response, so that the UI can display this list of available (undeleted) versions, allowing the user to pick a different version to launch.

This list of version objects (just the version label and ID) is also included in the app `details` and `ui` responses, for the same purpose.

And I agree that the doc strings should be updated to indicate the difference in these functions 👍

Reply:
Ah, that's the bit I was missing, that `list-app-versions` was just the version/version_id pairs (and not any sort of "complete" version -- even if it's only stuff from one table, more than just those two fields). Docstring updates and the dissoc you mention in the other comment both sound good, but now I see what I was missing in understanding this, haha. Looks good.